docs/background/architecture.md

```diff
@@ -4,4 +4,74 @@ sidebar_position: 4
 
 # Architecture
 
-*Pending.*
+This section describes the Guild’s current smart contract architecture.
+
+## Modules
+
+The DevTools Guild plans to follow the precedent set by the [Protocol Guild](https://protocol-guild.readthedocs.io/en/latest/03-onchain-architecture.html) with respect to the smart contract infrastructure it utilizes.
+
+Following are the two core contracts that DevTools Guild will utilize.
+
+- **The Splits contract** is used for splitting fund distributions to multiple addresses.
+- **The Vesting contract** is used to vest funds over a defined period of time before they can be released to the Splits contract.
+
+### Vesting Contract
+
+[Vesting Contract Repository](https://github.com/0xSplits/splits-vesting/tree/master)
+
+Vesting contracts enable the gradual release of funds to a designated beneficiary over a predetermined period. Each Vesting contract can manage multiple, isolated streams of tokens, all sharing the same vesting period but capable of starting and releasing independently.
+
+The process involves:
+
+- **Funding the Vesting Contract**: ETH and ERC20 tokens are sent to the Vesting contract, where they are held until a vesting stream is initiated.
+
+- **Starting a Vesting Stream**: A vesting stream for a specific token is initiated, beginning the linear vesting process over the set period. Multiple streams can operate simultaneously within the same contract.
+
+- **Releasing Vested Funds**: At any time, anyone can release the vested portion of tokens to the beneficiary. This mechanism ensures that funds are distributed gradually, aligning with predefined vesting schedules.
+
+#### How it works:
+- Donated tokens will accrue in a per-asset queue until a vesting stream is started for that batch of tokens by triggering the `startStream` function.
+    - This function is **permissionless**, meaning **any actor** can trigger it, regardless of whether they are a Guild member.
+- Once the vesting stream is started, the tokens will vest **linearly over 4 years** (or any predefined period set during contract deployment).
```

Comment:
We should say 12 months for the pilot instead of 4 years.

Reply:
updated